flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBKeyedStateBackend.java

```diff
@@ -249,12 +252,14 @@
 	 */
 	private RocksDBSerializedCompositeKeyBuilder<K> sharedRocksKeyBuilder;
 
+	private final RocksDbTtlCompactFiltersManager ttlCompactFiltersManager;
+
 	public RocksDBKeyedStateBackend(
 		String operatorIdentifier,
 		ClassLoader userCodeClassLoader,
 		File instanceBasePath,
 		DBOptions dbOptions,
-		ColumnFamilyOptions columnFamilyOptions,
+		Supplier<ColumnFamilyOptions> columnFamilyOptionsSupplier,
```

Comment:
Hmm, a bit off-topic but if we already need to create CFO per state, maybe it would be nice for user if this was a function `state name -> CFO` so that they can have their factory produce different setting in the CFO by state.